includes/updates/upgrade_3_0.php

```diff
@@ -70,6 +70,19 @@ function pmpro_upgrade_3_0_ajax() {
 	);
 	$subscriptions = PMPro_Subscription::get_subscriptions( $subscription_search_param );
 
+	// Check if the migration was successful.
+	$failed_migrations = array();
+	foreach ( $subscriptions as $subscription ) {
+		if ( empty( $subscription->get_billing_amount() ) && empty( $subscription->get_cycle_number() ) ) {
```

Comment:
```suggestion
		// This is the same check we used to get subs to update. We want to avoid infinite loops.
		if ( empty( $subscription->get_billing_amount() ) && empty( $subscription->get_cycle_number() ) ) {
```